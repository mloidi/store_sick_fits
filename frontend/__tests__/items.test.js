import ItemComponent from '../components/Item';
import { shallow, mount } from 'enzyme';
import formatMoney from '../lib/formatMoney';
import toJSON from 'enzyme-to-json';

const fakeItem = {
  id: 'AQW1234',
  title: 'My item',
  price: 2000,
  description: 'My description',
  image: 'picture.png',
  largeImage: 'largePicture.png'
};

describe('<Item/>', () => {
  it('renders and matches the sanpshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it('renders the image properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });

  // it('renders an displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const PriceTag = wrapper.find('PriceTag');
  //   expect(PriceTag.dive().text()).toBe(formatMoney(fakeItem.price));
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  // });
  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   console.log(buttonList.debug());
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  //   expect(buttonList.find('Link')).toBeTruthy;
  //   expect(buttonList.find('AddToCart').exists()).toBe(true);
  //   expect(buttonList.find('DeleteItem').exists()).toBe(true);
  // });
});
