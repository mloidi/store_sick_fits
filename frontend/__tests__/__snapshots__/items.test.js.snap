// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the sanpshot 1`] = `
<ItemStyles__Item>
  <img
    alt="My item"
    src="picture.png"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "AQW1234",
          },
        }
      }
    >
      <a>
        My item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $20
  </PriceTag>
  <p>
    My description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "AQW1234",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="AQW1234"
    />
    <DeleteItem
      id="AQW1234"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
